packages/instantsearch.js/src/components/InfiniteHits/InfiniteHits.tsx

```diff
@@ -58,15 +99,31 @@ const InfiniteHits = ({
 
   if (results.hits.length === 0) {
     return (
-      <Template
-        {...templateProps}
-        templateKey="empty"
-        rootProps={{
-          className: cx(cssClasses.root, cssClasses.emptyRoot),
-          onClick: handleInsightsClick,
-        }}
-        data={results}
-      />
+      <div
+        className={cx(cssClasses.root, cssClasses.emptyRoot)}
+        onClick={handleInsightsClick}
+      >
+        {banner &&
+          (templateProps.templates.banner ? (
+            <Template
+              {...templateProps}
+              templateKey="banner"
+              rootTagName="fragment"
+              data={{
+                banner,
+                className: cssClasses.bannerRoot,
+              }}
+            />
+          ) : (
+            <DefaultBanner banner={banner} classNames={cssClasses} />
+          ))}
+        <Template
+          {...templateProps}
+          templateKey="empty"
+          rootTagName="fragment"
+          data={results}
+        />
+      </div>
```

Comment:
The empty template fragment is being hoisted above the banner (even though the banner is above it in the DOM here). Any suggestions for forcing the expected layout (see Hits example in image below) without adding additional DOM nodes?

![image](https://github.com/algolia/instantsearch/assets/10552296/1e28770d-55c1-4805-b517-451ae6a0506c)

Reply:
@aymeric-giraudet or @Haroenv - any thoughts on this?

Reply:
That's odd @taylorcjohnson, is it related to CSS ?
How does it look like without our CSS stylesheets ? Maybe this should just be solved within `instantsearch.css`

Reply:
I don't believe so, @aymeric-giraudet - I removed the css to verify, but the DOM still shows a difference.

![image](https://github.com/algolia/instantsearch/assets/10552296/9acc2ae2-dbac-422a-a971-35fe777a13a0)

Reply:
I don't think it's the right solution, but what happens if you put the no results template in a div or give it a root tag name?

Reply:
Then it works as expected 😢 (either wrapping the empty template component in a `div` or changing the `rootTagName` to something else (like a `div`).

Reply:
Ah I think this may be due to the `RawHtml` component which is a hack for supporting Hogan with fragments, I'll check for a fix tomorrow morning :/

Reply:
Ah ok, I was hoping I was just missing some background context somewhere. If you can take a look I'd really appreciate it otherwise I can attempt as well.